lib/ansible/module_utils/facts/system/pkg_mgr.py

```diff
@@ -72,32 +78,22 @@ def _check_rh_versions(self, pkg_mgr_name, collected_facts):
         if os.path.exists('/run/ostree-booted'):
             return "atomic_container"
 
+        # default to dnf regardless to what was matched from PKG_MGRS, infer the default pkg_mgr below
+        pkg_mgr_name = 'dnf'
+        # since /usr/bin/dnf and /usr/bin/microdnf can point to different versions of dnf in different distributions
+        # the only way to infer the default package manager is to look at the binary they are pointing to
+        # microdnf is likely used only in fedora minimal container so /usr/bin/dnf takes precedence
+        for bin_path in ('/usr/bin/dnf', '/usr/bin/microdnf'):
```

Comment:
This assumption won't work. On Fedora 38, if both microdnf and dnf4 are installed, dnf5 will be used, which is incorrect. dnf4 is the default on Fedora 38.

Reply:
/usr/bin/dnf -> /usr/bin/dnf5 means that dnf5 is the default, but /usr/bin/microdnf -> /usr/bin/dnf5 does not. I can't think of a reasonable approach other than https://github.com/ansible/ansible/pull/80546/files#diff-8b63763b5cd493d8881baa72868750be9858f44056485d5ae0002de56994c009R81-R86 that handles using dnf4 on Fedora < 39 even if dnf5 is installed but works in the Fedora Minimal container.

Reply:
Okay, the current code makes more sense. It only checks microdnf if dnf is missing.

Reply:
/resolve